test/test_macros.hpp

```diff
@@ -11,19 +11,31 @@
 #include <cfloat>
 #include <math.h>
 
-inline float radians(float deg) {
-    return M_PI / 180.0f * deg;
+template<typename Type>
+inline Type radians(Type deg) {
+    return Type(M_PI) / Type(180.0) * deg;
 }
 
-inline float degrees(float rad) {
-    return rad / (M_PI / 180.0f);
+template<typename Type>
+inline Type degrees(Type rad) {
+    return rad / (Type(M_PI) / Type(180.0));
 }
 
 inline bool is_equal(float a, float b, float e) {
     return fabsf(a - b) < e;
 }
+
 inline bool is_equal(float a, float b) {
     return is_equal(a, b, FLT_EPSILON);
 }
 
```

Comment:
You should remove any non-template functions, when they get included multiple times in different object files it causes issues. Also, I think the function I added should work for this.